metabase/metabase
# fix interpolate missing values setting on stacked areas

Closes https://github.com/metabase/metabase/issues/38682

### Description

Fixes stacked areas with "Replace missing values with Linear interpolated" setting. Here is the visual explanation of how it works:
<img width="1725" alt="Screenshot 2024-03-12 at 5 22 53 PM" src="https://github.com/metabase/metabase/assets/14301985/4d601343-03c4-41ed-9490-701605a70b2e">

### How to verify

Ensure stacked area chart with "Replace missing values with Linear interpolated":
- interpolates area polygons when all series have nulls
- replaces nulls with zeros if at least one series in the dataset has non-null value

Example query
```
select 1 x, 10 y1, 20 y2, 30 y3
union all select 2, 10, null, 30
union all select 3, null, null, 30
union all select 4, null, null, null
union all select 5, 10, null, 30
```

### Demo

Before
<img width="1724" alt="Screenshot 2024-03-12 at 5 28 35 PM" src="https://github.com/metabase/metabase/assets/14301985/9823dc59-db93-400b-b549-58ca12f707b2">

After
<img width="1728" alt="Screenshot 2024-03-12 at 5 24 13 PM" src="https://github.com/metabase/metabase/assets/14301985/5dafd309-ca3b-4914-ad5f-5514c7471715">

DC.js
<img width="1725" alt="Screenshot 2024-03-12 at 5 22 44 PM" src="https://github.com/metabase/metabase/assets/14301985/64ead00f-e600-49ed-9f20-766d9bb7d7da">

### Checklist
